app/policies/application_policy.rb

```diff
@@ -26,7 +26,7 @@ def new?
   end
 
   def update?
-    user.present? && (record.public || user.admin? || record.user == user)
+    user.present? && (record.public || user.admin? || record.user == user || record.collaborators.pluck(:user_id).include?(user.id))
```

Comment:
`record.collaborators` assumes that `record` is a wiki. This will break for non-wiki objects such as user and collaboration. Wiki-specific policy code belongs in `wiki_policy.rb`.